java/src/main/java/com/powsybl/python/PyPowsyblApiHeader.java

```diff
@@ -64,6 +66,57 @@ private PyPowsyblApiHeader() {
         UnmanagedMemory.free(arrayPointer);
     }
 
+    @CStruct("network_metadata")
+    interface NetworkMetadataPointer extends PointerBase {
+
+        @CField("id")
+        CCharPointer getId();
+
+        @CField("id")
+        void setId(CCharPointer id);
+
+        @CField("name")
+        CCharPointer getName();
+
+        @CField("name")
+        void setName(CCharPointer name);
+
+        @CField("case_date")
+        CCharPointer getCaseDate();
+
+        @CField("case_date")
+        void setCaseDate(CCharPointer caseDate);
+
+        @CField("source_format")
+        CCharPointer getSourceFormat();
+
+        @CField("source_format")
+        void setSourceFormat(CCharPointer sourceFormat);
+
+        @CField("forecast_distance")
+        int getForecastDistance();
+
+        @CField("forecast_distance")
+        void setForecastDistance(int forecastDistance);
+    }
+
+    static NetworkMetadataPointer allocMainAttributePointer(Map<String, String> networkMetadata) {
+        NetworkMetadataPointer ptr = UnmanagedMemory.calloc(SizeOf.get(NetworkMetadataPointer.class));
+        ptr.setId(CTypeUtil.toCharPtr(networkMetadata.get("id")));
+        ptr.setName(CTypeUtil.toCharPtr(networkMetadata.get("name")));
+        ptr.setSourceFormat(CTypeUtil.toCharPtr(networkMetadata.get("sourceFormat")));
+        ptr.setForecastDistance(Integer.parseInt(networkMetadata.get("forecastDistance")));
+        ptr.setCaseDate(CTypeUtil.toCharPtr(networkMetadata.get("caseDate")));
+        return ptr;
+    }
+
+    static void freeMainAttributePointer(NetworkMetadataPointer networkMetadataPointer) {
+        UnmanagedMemory.free(networkMetadataPointer.getId());
```

Comment:
Be careful, maybe I am wrong but in all the other places in pypowsybl, I didn't released strings because I think this is done on python side when python string that wrap c side allocated mem are garbage collected.

Reply:
After some reading, it seems to me that python strings are always built by copying the input C string (see [python C API doc on strings](https://docs.python.org/3/c-api/unicode.html#creating-and-accessing-unicode-strings), and [pybind11](https://pybind11.readthedocs.io/en/stable/advanced/cast/strings.html#returning-c-strings-to-python) doc).

So maybe we have some memory leaks here!

Reply:
Note: if it's really released by python we could have some issues anyway, because graalvm doc states that memory allocated through the graalvm API needs to be deallocated the same way !

Reply:
I already tried to release string and it always end up with a crash

Reply:
So I did a test in order to confirm or not the leak, see #181 : it does confirm it! 
Probably the crashed you obtained were caused by something else ? (for example at first I had a "free(): invalid pointer" error because my deallocation code was wrong, I released the char** at adress i instead of the char*)

Reply:
Oups, probably, I don't remenber what was my use case, it was at the very early stage of the pypowsybl prototyping.
Thanks for the fix #181